### PR TITLE
feat(datagrid): add ability to unsort columns

### PR DESCRIPTION
**Pull request by valentin-mladenov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
Once turned on column sort can only be rotated between `ASC` and `DESC ` when clicking on the column, but can't be turned off. 

Issue Number: CDE-2416

## What is the new behavior?
Column sort can be turned off by clicking third time on the column.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by Jinnie — APPROVED** [review]
LGTM

**Comment by kevinbuhmann** [comment]
I would reword the PR title and commit message subject to "add ability to unsort columns".

File: projects/angular/src/data/datagrid/datagrid-column.ts [hunk 1]
```diff
@@ -357,6 +341,19 @@ export class ClrDatagridColumn<T = any>
...
+      let nextSort = ClrDatagridSortOrder.UNSORTED;
+
+      if (this._sortOrder === ClrDatagridSortOrder.UNSORTED) {
+        nextSort = ClrDatagridSortOrder.ASC;
+      } else if (this._sortOrder === ClrDatagridSortOrder.ASC) {
+        nextSort = ClrDatagridSortOrder.DESC;
+      }
+
+      this.doSort(nextSort);
+      return;
```
Comment (dtsanevmw) [inline projects/angular/src/data/datagrid/datagrid-column.ts:354]: Column button is using sort() only and because of the return here it would not emit events that we expose trough API  and also the other functions would not pass trough

**Comment by valentin-mladenov** [comment]
@kevinbuhmann `clrDgSortDisableUnsort` input is added in order to turn on/off the unsorted state of a column.
Does the name make sense?

@dtsanevmw can you review the changes we disscussed, separately?

**Comment by kevinbuhmann** [comment]
Has this gone through design? I'd like to see the proposed documentation/guidance before we release this. I'm not clear on why we need the ability to disable unsort. That might cause unnecessary UX differences in products. As a user I would not expect not to be able to unsort some grids/columns, but not all of them.

File: projects/demo/src/app/datagrid/sorting/sorting.html [hunk 2]
```diff
@@ -74,15 +74,21 @@ <h2>Custom sort</h2>
   <b>{{sortOrder === 1 ? " ascendingly" : sortOrder === -1 ? " descendingly" : ""}}</b>.
 </p>
 <p>
-  <button class="btn btn-secondary" (click)="sortOrder = 1" [disabled]="sortOrder === 1">Sort ascendingly</button>
-  <button class="btn btn-secondary" (click)="sortOrder = -1" [disabled]="sortOrder === -1">Sort descendingly</button>
-  <button class="btn btn-secondary" (click)="sortOrder = 0" [disabled]="sortOrder === 0">Clear sort</button>
+  <button class="btn btn-secondary" (click)="setSortOrder(1)" [disabled]="sortOrder === 1">Sort ascendingly</button>
+  <button class="btn btn-secondary" (click)="setSortOrder(-1)" [disabled]="sortOrder === -1">Sort descendingly</button>
+  <button class="btn btn-secondary" (click)="setSortOrder(0)" [disabled]="sortOrder === 0">Clear sort</button>
```
Comment (kevinbuhmann) [inline projects/demo/src/app/datagrid/sorting/sorting.html:79]: Why were these changed from direct assignment? The `setSortOrder` just sets the `sortOrder` property, so it is unnecessary and should be removed.

File: projects/demo/src/app/datagrid/sorting/sorting.ts [hunk 3]
```diff
@@ -23,10 +29,19 @@ export class DatagridSortingDemo {
...
     inventory.reset();
     this.users = inventory.all;
   }
+
+  setSortOrder(newOrder: ClrDatagridSortOrder) {
+    this.sortOrder = newOrder;
+  }
+
+  clrDgSortOrderChange($event: ClrDatagridSortOrder) {
+    console.log($event);
```
Comment (kevinbuhmann) [inline projects/demo/src/app/datagrid/sorting/sorting.ts:45]: I think this is leftover debugging code. It's only applies to one column, and the log message is not particularly helpful. Please remove.

**Review by kevinbuhmann — CHANGES_REQUESTED** [review]
See my comments.

File: projects/angular/src/data/datagrid/datagrid-column.ts [hunk 4]
```diff
@@ -202,16 +203,12 @@ export class ClrDatagridColumn<T = any>
...
-      if (comparator) {
-        this._sortBy = comparator;
-      } else {
-        if (this.field) {
-          this._sortBy = new DatagridPropertyComparator(this.field);
-        } else {
-          delete this._sortBy;
-        }
-      }
+      delete this._sortBy;
```
[the remark this answers is not in the document]
Comment (valentin-mladenov) [reply projects/angular/src/data/datagrid/datagrid-column.ts:211]: It was initally made that way. I just carried it over.

I followed the first principle, if it's working don't touch it.

File: projects/demo/src/app/datagrid/sorting/sorting.ts (hunk 3 above)
Comment (valentin-mladenov) [reply projects/demo/src/app/datagrid/sorting/sorting.ts]: corrected in commit 3684b86

File: projects/demo/src/app/datagrid/sorting/sorting.html (hunk 2 above)
Comment (valentin-mladenov) [reply projects/demo/src/app/datagrid/sorting/sorting.html]: corrected in commit 3684b86

File: projects/angular/src/data/datagrid/datagrid-column.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (valentin-mladenov) [reply projects/angular/src/data/datagrid/datagrid-column.ts:211]: Also is not in the scope of this change. I can open chore PR for it.

[the remark this answers is not in the document]
Comment (kevinbuhmann) [reply projects/angular/src/data/datagrid/datagrid-column.ts:211]: Ah, I see that now. Disregard.
